text/1121-extraneous-invisible-character-stripping.md

```diff
@@ -0,0 +1,230 @@
+        - indentation is determined by the smallest number of invisible characters for each line
+            - skip lines with 0 characters
+            - ideally all other lines have the same or greater indentation
+            - if spaces and tabs are mixed, we can ignore tabs -- we don't do any indentation stripping
+            - stripping tab indentation can only happen if all lines are indented with tabs
+
+
+### Examples
+
+> [!NOTE]
+> These examples use the format described in [RFC #931](https://github.com/emberjs/rfcs/pull/931) for brevity.
+
+> [!IMPORTANT]
+> Every invisible character and line break unneededly adds to the app-compiled wire-format, which leads to more runtime code for end-users to deal with. The more extraneous invisible characters we can eliminate, the better.
+> For _aggressive_ invisible character stripping, [ember-hbs-minifier](https://github.com/mainmatter/ember-hbs-minifier/) is still a viable option.
+
+<table>
+    <thead>
+<th>Input</th>
+<th>Output (today)</th>
+<th>Output (desired)</th>
+</thead>
+<tbody>
+<tr><td>
+
+
+```gjs
+<template>
+  <span>x</span>
+</template>
+```
+
+</td><td>
+
+```js
+export default template(
+    '\n  <span>x</span>\n'
+);
+```
+
+</td><td>
+
+```js
+export default template(
+    '<span>x</span>'
+)
+```
+
+</td></tr>
+<tr><td>
+
+```gjs
+const MyTemplate = <template>
+  <span>x</span>
+</template>
+
+<template>
+  <MyTemplate />
+</template>
+```
+
+</td><td>
+
+```js
+const MyTemplate = template(
+    '\n  <span>x</span>\n'
+);
+
+export default template(
+    '\n  <MyTemplate>\n',
+    { scope: () => ({ MyTemplate })}
+)
+```
+
+</td><td>
+
+```js
+const MyTemplate = template(
+    '<span>x</span>'
+);
+
+export default template(
+    '<MyTemplate>',
+    { scope: () => ({ MyTemplate })}
+)
+```
+
+</td></tr>            
+<tr><td>
+
+
+```gjs
+<template>
+  Hello
+  <span>there</span>.
+  <p>
+    <span>how are you</span>
+  </p>
+</template>
+```
+
+</td><td>
+
+```js
+export default template(
+    '\n  Hello\n  <span>there</span>.\n  <p>\n    <span>how are you</span>\n  </p>\n'
+);
+```
+
+</td><td>
+
+```js
+export default template(
+    'Hello <span>there</span>.\n<p>\n  <span>how are you</span>\n</p>'
+)
+```
+
+</td></tr>
+</tbody>
+</table>
+
+### Configuration
+
+There will be no configuration, as this can be enabled with 0 breaking changes to existing behavior.
+
+### Backward Compatibility
+
+- Existing applications using ember-hbs-minifier can continue to do so -- this RFC does not affect the behavior of those transforms.
+
+## How we teach this
+
+This feature requires minimal teaching as it operates transparently. It requires more teaching now to explain all the excessive indentation we have in gjs / gts.
+
+When releasing the next version of `content-tag`, and `ember-template-imports` should note the new behavior.
+
+In the template-tag section of the guide, add a sentence that says that indentation spacing is stripped.
```

Comment:
~~~suggestion
In the template-tag section of the guide, add a sentence that says that indentation spacing is stripped.

Also, we should add to the guides that to opt-out, folks can add a comment like this:

```gjs
    <template>
{{!-- prevent automatic de-indent --}}
       pre content here
    </template>
```
~~~